controller/rest/registry.go

```diff
@@ -179,6 +231,40 @@ func handlerRegistryCreate(w http.ResponseWriter, r *http.Request, ps httprouter
 		return
 	}
 
+	handlerRegistryConfig(data, w, r, ps)
+}
+
+func handlerRegistryConfigV2(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	body, _ := ioutil.ReadAll(r.Body)
+
+	var v2data api.RESTRegistryConfigDataV2
+	err := json.Unmarshal(body, &v2data)
+	if err != nil || v2data.Config == nil {
+		restRespError(w, http.StatusBadRequest, api.RESTErrInvalidRequest)
```

Comment:
access control checking should be done before REST server returns any error about the request/payload